sunpy/data/_sample.py

```diff
@@ -51,52 +65,58 @@
     "NORH_LIGHTCURVE": ("tca110810", "")
 }
 
-sample_files = {}
-for key in _files:
-    sample_files[key] = os.path.abspath(os.path.join(sampledata_dir, _files[key][0]))
+
+free_required=145227776
+if disk_usage('/').free > free_required :
```

Comment:
Agreed, this is the incorrect path to be checking for free space (since it may not even be on same drive even as the user's SunPy data directory).

Also, this whole code structure is wrong.  The definition of `download_sample_data` should not depend on whether the user has sufficient free space at the time of import.  If anything, `download_sample_data` performing this checking whenever it is called.